content/guides/code-splitting.md

```diff
@@ -1,41 +1,226 @@
+
+?> Update bash output.
+
+Here are some other useful plugins and loaders provide by the community for splitting code:
+
+- [`ExtractTextPlugin`](/plugins/extract-text-webpack-plugin): Useful for splitting CSS out from the main application.
+- [`bundle-loader`](/loaders/bundle-loader): Used to split code and lazy load the resulting bundles.
+- [`promise-loader`](https://github.com/gaearon/promise-loader): Similar to the `bundle-loader` but uses promises.
+
+
+## Dynamic Imports
+
+Two similar techniques are supported by webpack when it comes to dynamic code splitting. The first and more preferable approach is use to the [`import()` syntax](/api/module-methods#import-) that conforms to the [ECMAScript proposal](https://github.com/tc39/proposal-dynamic-import) for dynamic imports. The legacy, webpack-specific approach is to use [`require.ensure`](/api/module-methods#require-ensure). Let's try using the first of these two approaches...
+
+Before we start, let's remove the `entry` and `CommonsChunkPlugin` from our config as they won't be needed for this next demonstration:
+
+__webpack.config.js__
+
+``` diff
+  const path = require('path');
+- const webpack = require('webpack');
+
+  module.exports = {
+    entry: {
++     index: './src/index.js'
+-     index: './src/index.js',
+-     vendor: [
+-       'lodash'
+-     ]
+    },
+-   plugins: [
+-     new webpack.optimize.CommonsChunkPlugin({
+-       name: 'vendor' // Specify the common bundle's name.
+-     })
+-   ],
+    output: {
+      filename: '[name].bundle.js',
+      path: path.resolve(__dirname, 'dist')
+    }
+  };
+```
+
+Now, instead of statically importing lodash, we'll use dynamic importing to separate a chunk:
+
+__src/index.js__
+
+``` diff
+- import _ from 'lodash';
+-
+- function component() {
++ function getComponent() {
+-   var element = document.createElement('div');
+-
+-   // Lodash, now imported by this script
+-   element.innerHTML = _.join(['Hello', 'webpack'], ' ');
++   return import(/* webpackChunkName: "lodash" */ 'lodash').then(module => {
++     var element = document.createElement('div');
++
++     element.innerHTML = _.join(['Hello', 'webpack'], ' ');
++
++     return element;
++
++   }).catch(error => 'An error occurred while loading the component');
+  }
+
+- document.body.appendChild(component());
++ getComponent().then(component => {
++   document.body.appendChild(component);
++ })
+```
+
+Note the use of `webpackChunkName` in the comment. This will cause our separate bundle to be named `lodash.bundle.js` instead of just `[id].bundle.js`. For more information on `webpackChunkName` and the other available options, see the [`import()` documentation](/api/module-methods#import-). Let's run webpack to see lodash separated out to a separate bundle:
```

Comment:
I think I understand what's going on.

The **chunk** is named correctly, but the output isn't, because we removed it from the webpack config. It'll just generate an index-based name as far as I can tell.

So the output is correct, just this part of the guide isn't ;)